### PR TITLE
Parse depth

### PR DESCRIPTION
**Pull request by necryin**
add parseDepth flag for speed up generate
on my laptop right parseDepth flag give ~x3 speed up (1m -> 23sec)

### CODE REVIEW

File: parser.go [hunk 1]
```diff
@@ -159,6 +159,7 @@ func (parser *Parser) ParseAPI(searchDir string, mainAPIFile string) error {
 
 	var t depth.Tree
 	t.ResolveInternal = true
+	t.MaxDepth = parseDepth
```
Comment (sdghchj) [inline parser.go:162]: Can you help move the three lines into ``` if parser.ParseDependency {
}``` ?

Comment (necryin) [reply parser.go]: done

Comment (sdghchj) [reply parser.go]: thks

**Comment by easonlin404** [comment]
@sdghchj I would consider drop code coverage checks if needs so that you can merge PR.

File: cmd/swag/main.go [hunk 2]
```diff
@@ -67,13 +68,18 @@ var initFlags = []cli.Flag{
...
+		Name:  parseInternalFlag,
 		Usage: "Parse go files in internal packages, disabled by default",
 	},
 	&cli.BoolFlag{
-		Name:  "generatedTime",
+		Name:  generatedTimeFlag,
 		Usage: "Generate timestamp at the top of docs.go, true by default",
 	},
+	&cli.IntFlag{
+		Name:  parseDepthFlag,
```
Comment (easonlin404) [inline cmd/swag/main.go:79]: @necryin could you help pull a new PR to update  https://github.com/swaggo/swag#swag-cli as well ?

**Comment by sdghchj** [comment]
> @sdghchj I would consider drop code coverage checks if needs so that you can merge PR.

That wound be nice

**Pull request merged**